src/algebra/ring.lean

```diff
@@ -623,3 +623,24 @@ variables [domain α]
 end domain
 
 end units
+
+/-- A predicate to express that a commutative ring is an integral domain.
+
+This is mainly useful because such a predicate does not contain data,
+and can therefore be easily transported along ring isomorphisms. -/
+class is_integral_domain (R : Type u) [comm_ring R] : Prop :=
+(eq_zero_or_eq_zero_of_mul_eq_zero : ∀ x y : R, x * y = 0 → x = 0 ∨ y = 0)
+(zero_ne_one : (0 : R) ≠ 1)
```

Comment:
Not sure if this should be a class, or just a predicate. It will be rarely used.

Reply:
True. I also hope it will be rarely used (-;

Reply:
It's no longer a class.